app/services/document_security_service.rb

```diff
@@ -45,7 +45,20 @@ def build_cloudfront_url(attachment)
       s3_key = attachment.blob.key
       # Ensure DocuSeal prefix for document organization
       prefixed_key = s3_key.start_with?('docuseal/') ? s3_key : "docuseal/#{s3_key}"
-      "#{cloudfront_base_url}/#{prefixed_key}"
+
+      # Build base URL
+      base_url = "#{cloudfront_base_url}/#{prefixed_key}"
+
+      # Add Content-Disposition header to set the download filename
+      # This must be added before signing the URL
+      filename = attachment.blob.filename.to_s
+      disposition = "inline; filename=\"#{filename}\"; filename*=UTF-8''#{CGI.escape(filename)}"
+
+      # Add query parameters for response headers
+      escaped_disposition = CGI.escape(disposition)
+      escaped_content_type = CGI.escape(attachment.blob.content_type)
+      "#{base_url}?response-content-disposition=#{escaped_disposition}" \
+        "&response-content-type=#{escaped_content_type}"
```

Comment:
[Potentially Blocking]: All of this logic feels really important, we should probably add some tests that mock some of this behavior and assert against our expected outputs.